src/actions/querystringsearch/querystringsearch.js

```diff
@@ -10,23 +10,33 @@ import { settings } from '~/config';
 export function getQueryStringResults(path, data, subrequest, page) {
   // fixes https://github.com/plone/volto/issues/1059
 
+  let d = JSON.parse(JSON.stringify(data));
+  if (data?.depth != null) {
+    delete d.depth;
+    d.query.forEach((q) => {
+      if (q.i === 'path') {
+        q.v += '::' + data.depth;
+      }
+    });
+  }
+
   return {
     type: GET_QUERYSTRING_RESULTS,
     subrequest,
     request: {
       op: 'post',
       path: `${path}/@querystring-search`,
       data: {
-        ...data,
-        ...(!data.b_size && {
+        ...d,
+        ...(!d.b_size && {
           b_size: settings.defaultPageSize,
         }),
         ...(page && {
-          b_start: data.b_size
+          b_start: d.b_size
             ? data.b_size * (page - 1)
```

Comment:
Missing data -> d ?
I'm not a fan of using such short var names, I see that the naming here might be redundant, but still... Can we stick  to meaningful name vars?